ext/mixed/js/japanese.js

```diff
@@ -86,3 +85,29 @@ function jpDistributeFurigana(expression, reading) {
 
     return segmentize(reading, groups) || fallback;
 }
+
+function jpDistributeFuriganaInflected(expression, reading, source) {
+    const output = [];
+
+    let stemLength = 0;
+    const shortest = Math.min(source.length, expression.length);
+    const sourceHiragana = jpKatakanaToHiragana(source);
+    const expressionHiragana = jpKatakanaToHiragana(expression);
+    while (stemLength < shortest && sourceHiragana[stemLength] === expressionHiragana[stemLength]) {
+        ++stemLength;
+    }
+    const offset = source.length - stemLength;
+
+    for (const segment of jpDistributeFurigana(
+        source.slice(0, offset === 0 ? source.length : source.length - offset),
```

Comment:
For improved semantics, I would recommend doing something like:

```js
const expression2 = source.slice(0, offset === 0 ? source.length : source.length - offset);
const reading2 = reading.slice(0, offset === 0 ? reading.length : reading.length - expression.length + stemLength);
for (const segment of jpDistributeFurigana(expression2, reading2)) {
```

I think expression2 can also be simplified to
```js
const expression2 = source.slice(0, source.length - offset);
```